### PR TITLE
Feat: Cache npm dependencies

### LINKED ISSUE #1160: Feature Request: Cache npm dependencies
**Is your feature request related to a problem? Please describe.**
We can cache the `node_modules` folder in workflows with artifacts to ensure faster workflow runs. 

**Describe the solution you'd like**
[Refer to this guide](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows)

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by bhav-khurana**
**What kind of change does this PR introduce?**
Feature: Cache npm dependencies

**Issue Number:**
Fixes #1160 

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
Added feature to cache npm dependencies in .github/workflows

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide] (https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

File: .github/workflows/pull-requests.yml [hunk 1]
```diff
@@ -29,6 +29,23 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Install Dependencies
```
Comment (EshaanAgg) [inline .github/workflows/pull-requests.yml:31]: Rename to `Restore node_modules from cache`

File: .github/workflows/pull-requests.yml [hunk 2]
```diff
@@ -29,6 +29,23 @@ jobs:
...
+            node_modules
+          key: ${{ runner.os }}-generate-docs-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-generate-docs-${{ env.cache-name }}-
+            ${{ runner.os }}-generate-docs-
+            ${{ runner.os }}-
+
+      - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+        name: List the state of node modules
+        continue-on-error: true
```
Comment (EshaanAgg) [inline .github/workflows/pull-requests.yml:48]: We do not need to list the modules installed conditionally. Rename this step to `Install dependencies` and always run npm ci.

File: .github/workflows/pull-requests.yml [hunk 3]
```diff
@@ -71,8 +88,25 @@ jobs:
...
```
Comment (EshaanAgg) [inline .github/workflows/pull-requests.yml:109]: This is a different cache. The earlier one was `node_modules` of the app; this is for the global installation of `graphql-inspector`. I don't think so we need caching here. Please revert this.

**Comment by bhav-khurana** [comment]
@EshaanAgg the lines missing coverage were not part of my edited files. Please have a look

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -42,8 +61,26 @@ jobs:
     steps:
       - name: Checkout the Repository
         uses: actions/checkout@v3
-        
-      - name: Install Dependencies
+
+      - name: Cache node modules
```
Comment (EshaanAgg) [inline .github/workflows/push.yml:65]: Rename to `Restore node_modules from cache`

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -42,8 +61,26 @@ jobs:
...
+          restore-keys: |
+            ${{ runner.os }}-generate-docs-${{ env.cache-name }}-
+            ${{ runner.os }}-generate-docs-
+            ${{ runner.os }}-
+
+      - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+        name: List the state of node modules
+        continue-on-error: true
+        run: npm list
         run: yarn install --legacy-peer-deps
```
Comment (EshaanAgg) [inline .github/workflows/push.yml:84]: Remove the conditional check, rename it to `Install dependencies`, and change the command to `npm install --legacy-peer-deps`.

**Review by EshaanAgg — CHANGES_REQUESTED** [review]

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
You have used two install commands `npm ci` and `npm install --legacy-peer-deps` in all the caches. Please use only one, preferably `npm ci` and if it doesn't work then `npm i --legacy-peer-deps`

**Review by EshaanAgg — APPROVED** [review]
LGTM. Thanks.

**Review by noman2002 — APPROVED** [review]

**Pull request merged**